core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -284,9 +286,9 @@ protected boolean removeEldestEntry(Map.Entry<Sha256Hash, Transaction> eldest) {
     protected volatile WalletFiles vFileManager;
     // Object that is used to send transactions asynchronously when the wallet requires it.
     protected volatile TransactionBroadcaster vTransactionBroadcaster;
-    // UNIX time in seconds. Money controlled by keys created before this time will be automatically respent to a key
+    // Money controlled by keys created before this time will be automatically respent to a key
     // that was created after it. Useful when you believe some keys have been compromised.
-    private volatile long vKeyRotationTimestamp;
+    private volatile Optional<Instant> vKeyRotationTime = Optional.empty();
```

Comment:
This could be `Instant.MAX` by default if we want to avoid using `Optional` altogether.